core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticator.java

```diff
@@ -71,6 +75,32 @@ protected Optional<Identity> createIdentity(String token)
                 .build());
     }
 
+    private void validateAudience(Claims claims)
+    {
+        Object tokenAudience = claims.get(AUDIENCE);
```

Comment:
change order, if `requiredAudience` is empty then getting audience from claims is redundant